core/src/test/java/org/mule/management/stats/ComponentStatisticsTestCase.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.management.stats;
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.*;
+import org.mule.tck.junit4.AbstractMuleTestCase;
+
+/**
+ * Validates some basic assumptions about the ComponentStatistics class
+ * behavior.
+ */
+public class ComponentStatisticsTestCase extends AbstractMuleTestCase {
+
+    /* This class was added to address MULE-6417 */
+    
+    private String env_statIntervalTime;
+    
+    @Before
+    public void setUp() {
+        // ComponentStatistics reads statIntervalTime from the environment.
+        // We will want to control it.
+        env_statIntervalTime = System.getProperty("statIntervalTime");
+        System.clearProperty("statIntervalTime");
+    }
+    
+    @After
+    public void tearDown() {
+        if (env_statIntervalTime != null) {
+            System.setProperty("statIntervalTime", env_statIntervalTime);
+        }
+    }
+    
+    private void assertValues(ComponentStatistics stats,
+            long numEvents, long totalTime, long avgTime, 
+            long maxTime, long minTime) {
+        assertEquals("getExecutedEvents", numEvents, stats.getExecutedEvents());
```

Comment:
Please change these assertions to use hamcrest: assertEquals to assertThat(..., is())